Visualizer/src/components/dashboard/Heatmap.vue

```diff
@@ -105,10 +105,14 @@ const updateState = () => {
         for (let y = min[props.dimY]; y <= max[props.dimY]; y++) {
           const dim1 = Math.floor(x / dimXlength);
           const dim2 = Math.floor(y / dimYlength);
-          if (dim2 >= series.length || dim1 >= series[0].data.length) {
+          if (dim2 >= series.length || dim1 >= series[0].data.length || dim2 < 0 || dim1 < 0) {
             continue;
           }
-          series[dim2].data[dim1] += 1;
+          if (series[dim2]) {
+            series[dim2].data[dim1] += 1;
+          } else {
+            console.log(series, dim2, dim1);
+          }
```

Comment:
hmmm this seems like a workaround. Why can the dimensions be sub-zero now and why might series[dim2] be non-existent?

Reply:
space agents are allowed to reserve space outside the field, the other was a hack to fix this issue